### PR TITLE
Support multiple-packages 

### PR DESCRIPTION
**Pull request by haya14busa**
ref: https://github.com/mattn/goveralls/issues/68 https://github.com/mattn/goveralls/issues/20

I created branch based on de5a843 to support run coverage for multiple packages.

de5a843 has some problem and this p-r fixes them.

### What this p-r did
- support `-coverpkgs` arg to run coverage for all packages, not only for packages to run test.
  - example: https://github.com/haya14busa/go-cover-multi-pkgs-example/pull/1
- support `-package` arg for backword compatibility
  - example: https://coveralls.io/builds/7837950 https://github.com/haya14busa/go-cover-multi-pkgs-example/pull/3#issuecomment-246214508
- merge coverprofile's into one profile before sending to coveralls.
  - just sending all coverprofiles turns coverage to 100%, but sourcefiles contains duplicated files. https://github.com/haya14busa/go-cover-multi-pkgs-example/pull/2#issuecomment-246208492
- do not allow extra arguments https://github.com/mattn/goveralls/commit/1fe178dd7c29c51477cb12165777a3bd151e73f6 for compatibility.
  - Now, we cannot handle `goveralls pkg`. The users should use `-package` arg instead (`goveralls -package pkg`). `TOKEN` arg seems deprecated and the users should use `-repotoken`.
- ignore vendor directories.

working example: https://github.com/haya14busa/go-cover-multi-pkgs-example/pull/3

I guess I should write more tests, but it works perfectly for me.

I read https://github.com/golang/go/issues/6909#issuecomment-233493644 issue and tries to write a patch to golang at first,
but we cannot support `-coverprofile` with multiple packages without drastic interface changes (like changes -coverprofile format, etc..), imo.

So, I decided to crate this p-r.

What do you think?

### CODE REVIEW

File: gocover.go [hunk 1]
```diff
@@ -42,7 +42,10 @@ func mergeProfs(pfss [][]*cover.Profile) []*cover.Profile {
 	ret := make([]*cover.Profile, 0, len(head))
 	for i, profile := range head {
 		for _, ps := range rest {
-			if len(ps) < i+1 {
+			if len(ps) < 1 {
```
Comment (mattn) [inline gocover.go]: == 0

File: gocover.go [hunk 2]
```diff
@@ -33,6 +33,13 @@ func findFile(file string) (string, error) {
 // mergeProfs merges profiles for same target packages.
 // It assumes each profiles have same sorted FileName and Blocks.
 func mergeProfs(pfss [][]*cover.Profile) []*cover.Profile {
+	// skip empty profiles ([no test files])
+	for i := 0; i < len(pfss); i++ {
+		if len(pfss[i]) < 1 {
```
Comment (mattn) [inline gocover.go]: == 0

File: gocover.go [hunk 3]
```diff
@@ -33,6 +33,13 @@ func findFile(file string) (string, error) {
...
+			continue
+		}
+		pfss = pfss[i:]
+	}
 	if len(pfss) < 1 {
```
Comment (mattn) [inline gocover.go]: please fix this too (this is mine)

**Comment by mattn** [comment]
LGTM :muscle: 

Could you please squash commits?

**Comment by haya14busa** [comment]
Thanks! squashed  done and add test for `mergeProfs()` 23dab33

**Comment by mattn** [comment]
:+1: Thanks.

**Comment by haya14busa** [comment]
Thank you for including it :sparkles: :bird:

**Pull request merged**